### ISSUE TITLE
possibility to run envoy as daemon

### ISSUE THREAD

**Issue by wxmeng04** [envoyproxy/envoy#29834]
**If you are reporting *any* crash or *any* potential security issue, *do not*
open an issue in this repo. Please report the issue via emailing
envoy-security@googlegroups.com where the issue will be triaged appropriately.**

*Title*: *possibility to run envoy as daemon*

*Description*:
> As described in official document https://www.envoyproxy.io/docs/envoy/latest/start/quick-start/run-envoy.html, it's possible to run envoy as a daemon, but actually it's not. Taking redis as an example, if you run redis like below and specify daemonize as `yes`, the redis main process will be run at background without blocking the command line. If you run envoy as per the official example, the main process will occupy the command line.
If the project is also aimed to be a general purpose gateway, it's better envoy can be started as a real daemon. In this case it's possible to specify `type` as `forking` and declare a `pidfile` in the systemd unit file like other services normally do.

```
┌──(root💀ep0001)-[/etc/redis]
└─# redis-server ./redis.conf 
┌──(root💀ep0001)-[/etc/redis]
└─# 

┌──(root💀ep0001)-[~]
└─# ./envoy-contrib-1.27.0-linux-x86_64 -c ./envoy-demo.yaml 
[2023-09-27 15:43:20.129][16626][info][main] [source/server/server.cc:413] initializing epoch 0 (base id=0, hot restart version=11.104)
[2023-09-27 15:43:20.129][16626][info][main] [source/server/server.cc:415] statically linked extensions:
...

[2023-09-27 15:43:20.713][16626][info][upstream] [source/common/upstream/cluster_manager_impl.cc:226] cm init: all clusters initialized
[2023-09-27 15:43:20.713][16626][info][main] [source/server/server.cc:918] all clusters initialized. initializing init manager
[2023-09-27 15:43:20.713][16626][info][config] [source/extensions/listener_managers/listener_manager/listener_manager_impl.cc:858] all dependencies initialized. starting workers
[2023-09-27 15:43:20.719][16626][warning][main] [source/server/server.cc:814] there is no configured limit to the number of allowed active connections. Set a limit via the runtime key overload.global_downstream_max_connections
|
```
[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by alyssawilk** [comment]
AFIK that capability does not exist today.  Might be easier to have a wrapper scrpit do it but I suppose if you wanted to make it an option for hot restart it'd be plausible?

**Comment by wxmeng04** [comment]
> AFIK that capability does not exist today. Might be easier to have a wrapper scrpit do it but I suppose if you wanted to make it an option for hot restart it'd be plausible?

Yes, it's possible to run it as deamon as stated by the official documentation, but with python script. Compared to other linux daemons such as redis, nginx, postgresql, mysql and so on, this is not a usual case. I suppose this can be impoved latter. Or may I understand this project is only purposed for usage with containers and deamon is not important at all?

**Comment by Lyamc** [comment]
I struggled yesterday trying to get the systemd service to start at all, it never actually output anything to stdout no matter what I did so I'm not sure what was failing. I'm in an unpriviledged proxmox lxc container. I also tried the helper scripts but no luck.

Here's how I got it working. Note that I already set up the user and permissions in advanced,

```
[Unit]
Description=Envoy Proxy
After=network.target

[Service]
Type=forking
User=root
ExecStart=tmux new-session -d -s envoy 'sudo -u envoy envoy --config-path /etc/envoy/envoy.yaml'
ExecStop=tmux kill-session -t envoy
Restart=on-failure
RestartSec=10s
LimitNOFILE=65536

[Install]
WantedBy=multi-user.target
```

**Comment by phlax** [comment]
thanks, this is useful - ive been thinking to add a systemd setup to our debs
